src/SessionSetup/PersistentStorage.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SessionSetup/PersistentStorage.h"
+
+#include <QSettings>
+
+#include "OrbitGgp/Project.h"
+
+namespace orbit_session_setup {
+
+constexpr const char* kSelectedProjectIdKey{"kSelectedProjectIdKey"};
+constexpr const char* kSelectedProjectDisplayNameKey{"kSelectedProjectDisplayNameKey"};
+
+using orbit_ggp::Project;
+
+[[nodiscard]] std::optional<orbit_ggp::Project> LoadProjectFromPersistentStorage() {
+  QSettings settings;
+
+  std::optional<Project> project;
+  if (settings.contains(kSelectedProjectIdKey)) {
+    project = Project{
+        settings.value(kSelectedProjectDisplayNameKey).toString() /* display_name */,
+        settings.value(kSelectedProjectIdKey).toString() /* id */
+    };
+  }
+  return project;
+}
+
+void SaveProjectToPersistentStorage(std::optional<Project> project) {
```

Comment:
Regarding naming, I would prefer this (and "Load.." as well) to reflect the fact that this is used to remember the selected project. The naming does not suggest this at all, and then it's especially confusing that you can pass a nullopt in here to clear a stored project. This could also be pointed out in comments to the method, but I prefer descriptive naming.

Reply:
I named the Load method `LoadLastSelectedProjectFromPersistentStorage` and hope this is more descriptive.